libs/@local/hash-graph-sdk/typescript/src/entity.ts

```diff
@@ -463,6 +470,32 @@ export const flattenPropertyMetadata = (
   return flattened;
 };
 
+export const getClosedMultiEntityTypesFromResponse = (
+  response: GetEntitySubgraphResponse | GetEntitiesResponse,
+  entityTypesIds: [VersionedUrl, ...VersionedUrl[]],
+): ClosedMultiEntityType | undefined => {
+  if (!response.closedMultiEntityTypes) {
+    return;
+  }
+
+  const [firstEntityTypeId, ...restEntityTypesIds] = entityTypesIds.sort();
```

Comment:
I'm not sure it really matters here, but to note that `sort` mutates the original array, we can use this newish alternative to avoid mutating the input array
```suggestion
  const [firstEntityTypeId, ...restEntityTypesIds] = entityTypesIds.toSorted();
```

Reply:
This is one of the reasons why I like Rust, if there is a side effect it's impossible to have an existing reference to `entityTypeIds`, so it's possible to just sort the array and profit from it later again if we call `sort()` again (I assume `O(n)` for sorted arrays).
I changed it to `toSorted` but had to add `!` to `firstEntityTypeId` as it does not retain the exact type of the array.